src/targets/docker.ts

```diff
@@ -132,8 +166,14 @@ export class DockerTarget extends BaseTarget {
    */
   public async publish(version: string, revision: string): Promise<any> {
     await this.login();
-    await this.pull(revision);
-    await this.push(revision, version);
+
+    if (await this.hasBuildKit()) {
```

Comment:
I'm inclined to just _require_ buildkit and clean up the old code -- it's been around for half a decade

Reply:
Unfortunately that will fail on OSX, at least it does on my colima install.

Reply:
what does `docker version` give you? I'd be pretty surprised if it's older than 2017

Reply:
```
└─ % docker version
Client: Docker Engine - Community
 Version:           27.0.3
 API version:       1.43 (downgraded from 1.46)
 Go version:        go1.22.4
 Git commit:        7d4bcd863a
 Built:             Fri Jun 28 14:56:30 2024
 OS/Arch:           darwin/arm64
 Context:           colima

Server:
 Engine:
  Version:          24.0.5
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.7
  Git commit:       24.0.5-0ubuntu1
  Built:            Wed Aug 16 21:32:36 2023
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.7.2
  GitCommit:        
 runc:
  Version:          1.1.7-0ubuntu2.2
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:        
```

```
└─ % docker buildx version
docker: 'buildx' is not a docker command.
See 'docker --help'
```

Reply:
on macos you'll need the `docker-buildx` package (it's a plugin -- that's usually shipped with core but is separate for some reason there)

Reply:
yeah I think it's reasonable -- arguably the non-buildx behaviour is broken

Reply:
Updated 👍